flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/DeleteOrphanFiles.java

```diff
@@ -0,0 +1,351 @@
+ * under the License.
+ */
+package org.apache.iceberg.flink.maintenance.api;
+
+import java.time.Duration;
+import java.util.Map;
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
+import org.apache.flink.util.OutputTag;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.MetadataTableType;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.actions.DeleteOrphanFiles.PrefixMismatchMode;
+import org.apache.iceberg.flink.maintenance.operator.DeleteFilesProcessor;
+import org.apache.iceberg.flink.maintenance.operator.FileNameReader;
+import org.apache.iceberg.flink.maintenance.operator.FileUriKeySelector;
+import org.apache.iceberg.flink.maintenance.operator.ListFileSystemFiles;
+import org.apache.iceberg.flink.maintenance.operator.ListMetadataFiles;
+import org.apache.iceberg.flink.maintenance.operator.MetadataTablePlanner;
+import org.apache.iceberg.flink.maintenance.operator.OrphanFilesDetector;
+import org.apache.iceberg.flink.maintenance.operator.SkipOnError;
+import org.apache.iceberg.flink.maintenance.operator.TaskResultAggregator;
+import org.apache.iceberg.flink.source.ScanContext;
+import org.apache.iceberg.relocated.com.google.common.base.Splitter;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+/** Delete orphan files from the file system. */
+public class DeleteOrphanFiles {
+
+  private static final Schema FILE_PATH_SCHEMA = new Schema(DataFile.FILE_PATH);
+  private static final ScanContext FILE_PATH_SCAN_CONTEXT =
+      ScanContext.builder().streaming(true).project(FILE_PATH_SCHEMA).build();
+  private static final Splitter COMMA_SPLITTER = Splitter.on(",");
+
+  @Internal
+  public static final OutputTag<Exception> ERROR_STREAM =
+      new OutputTag<>("error-stream", TypeInformation.of(Exception.class));
+
+  static final String PLANNER_TASK_NAME = "Table Planner";
+  static final String READER_TASK_NAME = "Files Reader";
+  static final String FILESYSTEM_FILES_TASK_NAME = "Filesystem Files";
+  static final String METADATA_FILES_TASK_NAME = "List metadata Files";
+  static final String DELETE_FILES_TASK_NAME = "Delete File";
+  static final String AGGREGATOR_TASK_NAME = "Orphan Files Aggregator";
+  static final String FILTER_FILES_TASK_NAME = "Filter File";
+  static final String SKIP_ON_ERROR_TASK_NAME = "Skip On Error";
+
+  public static DeleteOrphanFiles.Builder builder() {
+    return new DeleteOrphanFiles.Builder();
+  }
+
+  private DeleteOrphanFiles() {
+    // Do not instantiate directly
+  }
+
+  public static class Builder extends MaintenanceTaskBuilder<DeleteOrphanFiles.Builder> {
+    private String location = null;
```

Comment:
NIT: No need to initialize
```suggestion
    private String location;
```

Reply:
Fix it !